libmattermost.c

```diff
@@ -1354,30 +1354,76 @@ mm_info_response(MattermostAccount *ma, JsonNode *node, gpointer user_data)
         const gchar *first_name;
         const gchar *last_name;
         const gchar *email;
+		const gchar *username;
+		const gchar *user_id;
+		const gchar *roles;
+
+		MattermostUser *mm_user = g_new0(MattermostUser, 1);
 
         nickname = json_object_get_string_member(user, "nickname");
         if (nickname && *nickname) {
                 purple_notify_user_info_add_pair_plaintext(user_info,"Nickname", nickname);
+				mm_user->nickname = g_strdup(nickname);	
         }
 
         first_name = json_object_get_string_member(user, "first_name");
         if (first_name && *first_name) {
                 purple_notify_user_info_add_pair_plaintext(user_info,"First Name", first_name);
+				mm_user->first_name = g_strdup(first_name);
         }
 
         last_name = json_object_get_string_member(user, "last_name");
         if (last_name && *last_name) {
                 purple_notify_user_info_add_pair_plaintext(user_info,"Last Name", last_name);
+				mm_user->last_name = g_strdup(last_name);
         }
 
         email = json_object_get_string_member(user, "email");
         if (email && *email) {
                 purple_notify_user_info_add_pair_plaintext(user_info,"Email address", email);
+				mm_user->email = g_strdup(email);
         }
 
+		username = json_object_get_string_member(user, "username");
+		if (username && *username) {
+                purple_notify_user_info_add_pair_plaintext(user_info,"Username", username);

```

Comment:
When specifying keys in the user_info callbacks, its important to use the `_()` function for localisation with gettext in the future, eg `_("Username")`